src/world/structure.lua

```diff
@@ -424,43 +412,38 @@ function Structure:command(dt)
 		if order == "shoot" then table.insert(gunOrders, order) end
 	end
 
+	gunControls = Gun.process(gunOrders)
+	engineControls = Engine.process(engineOrders)
+
 	local function create(object, location)
 		location = StructureMath.sumVectors(location, object[2])
 		object[2] = self:getWorldLocation(location)
 		table.insert(self.events.create, object)
 	end
 
-	gunControls = Gun.process(gunOrders)
-
-	for gun, t in pairs(self.guns) do
-		local function getPart(location)
-			location[3] = 0
-			local x, y = unpack(StructureMath.sumVectors(t[1], location))
-			return self.gridTable:index(x, y)
-		end
 
-		local newObject = gun:update(dt, shoot, getPart)
-		if newObject then
-			create(newObject, t[1])
-		end
+	local function getPart(location, pointer)
+		pointer[3] = 0
+		local x, y = unpack(StructureMath.sumVectors(location, pointer))
+		return self.gridTable:index(x, y)
 	end
 
-	engineControls = Engine.process(engineOrders)
-
-	for engine, t in pairs(self.engines) do
-		engine:update(self.body, t[1], engineControls)
-	end
 
-	for heal, t in pairs(self.heal) do
-		heal:update(dt)
-	end
+	local moduleInputs = {
+		dt = dt,
+		body = self.body,
+		getPart = getPart,
+		controls = {gun = gunControls, engine = engineControls}}
```

Comment:
```lua
	local moduleInputs = {
		dt = dt,
		body = self.body,
		getPart = getPart,
		controls = {gun = gunControls, engine = engineControls}}
```

I like this. It's a declarative description of the data that gets sent to modules.